### PR TITLE
Minor Adjustments to the "Brain Tumor" Quirk

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Brain Tumor damage is now paused when buckled onto a stasis bed. 

Having Mannitol metabolizing inside of you will also pause Brain Tumor damage. 

Lastly, crew members with brain tumors will now spawn with a pill bottle containing 4 x 5u Mannitol pills.

Also cuts some magic strings from the file upon request.

## Why It's Good For The Game

For the stasis tweak: This just makes sense, and is probably how it should've worked in the first place. It also gives brain tumor players a way to AFK without risking horrible traumas or death.

For the Mannitol tweak: This doesn't change much in the grand scheme of things - 5u of Mannitol will "heal" an additional 2.5 brain damage (25 damage instead of 22.5). This just means you're healed by an even 2 brain damage/tick instead of 1.8 due to the tumor.

For the pill bottle tweak: First, some stats. (These are also included in the comments of the file.)

- It takes 16 minutes, 40 seconds to die from brain damage.
- It takes 1 minute, 40 seconds to take 10% (20 organ damage) brain damage.
- 5u Mannitol will heal 25 brain damage (with the change above)

The 20u of Mannitol supplied will, if used to their **maximum potential**, extend the time it takes to die from brain failure to ~25 minutes. Hopefully, this allows both the medbay staff and the crew members with brain tumors to settle into the shift better before needing to secure a supply of Mannitol. Alternatively, they could save the pill bottle for emergencies. It gives the player options, which is a good thing.

## Changelog
:cl: Melbert
tweak: Brain Tumor damage is now paused when in stasis.
tweak: Brain Tumor damage is now paused when having Mannitol metabolizing in you.
add: Brain Tumor quirk users will now spawn with a pill bottle containing 4 x 5u Mannitol pills.
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
I was concerned that this PR wasn't going to live up to its title but then it totally did.

**Comment by tralezab** [comment]
Really well thought through, i like it a lot. good work!

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -53,17 +53,48 @@
...
+	/* A couple of brain tumor stats for anyone curious / looking at this quirk for balancing:
+	 * - It takes less 16 minute 40 seconds to die from brain death due to a brain tumor.
+	 * - It takes 1 minutes 40 seconds to take 10% (20 organ damage) brain damage.
+	 * - 5u mannitol will heal 12.5% (25 organ damage) brain damage
+	 */
 /datum/quirk/brainproblems
 	name = "Brain Tumor"
 	desc = "You have a little friend in your brain that is slowly destroying it. Better bring some mannitol!"
 	value = -3
+	var/where
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm:65]: below inherited and document please.

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -53,17 +53,48 @@
...
+/datum/quirk/brainproblems/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/pills = new /obj/item/storage/pill_bottle/mannitol/braintumor()
+	var/list/slots = list(
+		"in your left pocket" = ITEM_SLOT_LPOCKET,
+		"in your right pocket" = ITEM_SLOT_RPOCKET,
+		"in your backpack" = ITEM_SLOT_BACKPACK,
+		"in your hands" = ITEM_SLOT_HANDS
+	)
+	where = H.equip_in_one_of_slots(pills, slots, FALSE) || "at your feet"
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm:80]: || compares so it will either return 0/1

File: code/datums/traits/negative.dm [hunk 3]
```diff
@@ -53,17 +53,48 @@
...
+
+/datum/quirk/brainproblems/post_add()
+	var/mob/living/carbon/human/H = quirk_holder
+	if(where == "in your backpack")
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm:84]: magic string

File: code/datums/traits/negative.dm [hunk 4]
```diff
@@ -53,17 +53,48 @@
...
+
+	to_chat(quirk_holder, "<span class='boldnotice'>There is a bottle of mannitol pills [where] to keep you alive until you can secure a supply of medication. Don't rely on it too much!</span>")
+
 /datum/quirk/brainproblems/on_process()
-	quirk_holder.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.2)
+	//If we're frozen, why does our brain hurt?
+	if(IS_IN_STASIS(quirk_holder)) 
+		return
+	//Having mannitol in you will also pause the brain damage (so it heals an even 2 brain damage instead of 1.8)
+	else if(quirk_holder.reagents.has_reagent(/datum/reagent/medicine/mannitol, needs_metabolizing = TRUE))
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm:94]: trait-bind it instead please

File: code/datums/traits/negative.dm [hunk 5]
```diff
@@ -53,17 +53,48 @@
...
+		return
+	else
+		quirk_holder.adjustOrganLoss(ORGAN_SLOT_BRAIN, 0.2)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm:97]:
```DM
if(a)
    return
if(b)
    return
c

File: code/datums/traits/negative.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/traits/negative.dm:80]: the other traits that spawn the player with items use the same line, so I assume it works?

File: code/datums/traits/negative.dm (hunk 3 above)
Comment (MrMelbert) [reply code/datums/traits/negative.dm:84]: the other traits use the same 'magic string', should I replace them all with a define?

File: code/datums/traits/negative.dm (hunk 4 above)
Comment (MrMelbert) [reply code/datums/traits/negative.dm:94]: I thought is_in_stasis was the preferred helper for stasis, unless i'm completely misunderstanding your request here

**Comment by MrMelbert** [comment]
@ExcessiveUseOfCobblestone honestly not entirely sure what you requested (kinda a brainlet) but I think this is what you meant by trait, please correct me if i'm off

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 6]
```diff
@@ -793,6 +793,15 @@
 	C.adjustOrganLoss(ORGAN_SLOT_BRAIN, -2*REM)
 	..()
 
+//Having mannitol in you will pause the brain damage from brain tumor (so it heals an even 2 brain damage instead of 1.8)
+/datum/reagent/medicine/mannitol/on_mob_metabolize(mob/living/carbon/C)
+	..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:798]:
```suggestion
	. = ..()
```

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 7]
```diff
@@ -793,6 +793,15 @@
...
+	ADD_TRAIT(C, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
+
+/datum/reagent/medicine/mannitol/on_mob_end_metabolize(mob/living/carbon/C)
+	REMOVE_TRAIT(C, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
+	..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:803]:
```suggestion
	. = ..()
```

**Pull request merged**